### ISSUE TITLE
task_run_id changes between mapped tasks in Cloud but not local

### ISSUE THREAD

**Issue by bnaul** [PrefectHQ/prefect#5528]
## Description

Title pretty much says it all; it's not the end of the world (you can make a unique ID if you need one by appending `map_index` in the style of `task_full_name`), but it seems like the behavior ought to be consistent.

## Reproduction

```
# flow.py
import prefect
from prefect import task, Flow


@task(log_stdout=True)
def print_run_id(i):
    print(f"{prefect.context['task_run_id']=}, {prefect.context['map_index']=}")


with Flow("mapped task run ID") as flow:
    print_run_id.map([0, 1])

if __name__ == "__main__":
    flow.run()
```
Local run:
```
(.venv) ➜  model git:(eurotr) ✗ python flow.py
[2022-03-10 19:44:28-0500] INFO - prefect.FlowRunner | Beginning Flow run for 'mapped task run ID'
[2022-03-10 19:44:28-0500] INFO - prefect.TaskRunner | Task 'print_run_id': Starting task run...
[2022-03-10 19:44:28-0500] INFO - prefect.TaskRunner | Task 'print_run_id': Finished task run for task with final state: 'Mapped'
[2022-03-10 19:44:28-0500] INFO - prefect.TaskRunner | Task 'print_run_id[0]': Starting task run...
[2022-03-10 19:44:28-0500] INFO - prefect.TaskRunner | prefect.context['task_run_id']='e9af0483-2207-45e5-abee-98b70d252681', prefect.context['map_index']=0
[2022-03-10 19:44:28-0500] INFO - prefect.TaskRunner | Task 'print_run_id[0]': Finished task run for task with final state: 'Success'
[2022-03-10 19:44:28-0500] INFO - prefect.TaskRunner | Task 'print_run_id[1]': Starting task run...
[2022-03-10 19:44:28-0500] INFO - prefect.TaskRunner | prefect.context['task_run_id']='e9af0483-2207-45e5-abee-98b70d252681', prefect.context['map_index']=1
[2022-03-10 19:44:28-0500] INFO - prefect.TaskRunner | Task 'print_run_id[1]': Finished task run for task with final state: 'Success'
[2022-03-10 19:44:28-0500] INFO - prefect.FlowRunner | Flow run SUCCESS: all reference tasks succeeded
```
Cloud run:
```
(.venv) ➜  model git:(eurotr) ✗ prefect register -p flow.py --project Sandbox
prefect run --project Sandbox --name 'mapped task run ID' --execute
Collecting flows...
Processing 'flow.py':
  Building `Local` storage...
  Registering 'mapped task run ID'... Done
  └── ID: 92ae2b99-9eb7-4b6c-b02a-b8b0dc4270e5
  └── Version: 1
======================== 1 registered ========================
Looking up flow metadata... Done
Creating run for flow 'mapped task run ID'... Done
└── Name: elegant-mink
└── UUID: 017c2675-3dec-44db-bbd3-59685a9cf639
└── Labels: ['agentless-run-778f3226']
└── Parameters: {}
└── Context: {}
└── URL: https://cloud.prefect.io/replica/flow-run/017c2675-3dec-44db-bbd3-59685a9cf639
Executing flow run...
└── 19:44:37 | INFO    | Creating subprocess to execute flow run...
└── 19:44:38 | INFO    | Beginning Flow run for 'mapped task run ID'
└── 19:44:39 | INFO    | Task '[0, 1]': Starting task run...
└── 19:44:40 | INFO    | Task '[0, 1]': Finished task run for task with final state: 'Success'
└── 19:44:41 | INFO    | Task 'print_run_id': Starting task run...
└── 19:44:41 | INFO    | Task 'print_run_id': Finished task run for task with final state: 'Mapped'
└── 19:44:41 | INFO    | Task 'print_run_id[0]': Starting task run...
└── 19:44:41 | INFO    | prefect.context['task_run_id']='4c906ec9-f639-4fd4-9ddd-8eec26a8b614', prefect.context['map_index']=0
└── 19:44:41 | INFO    | Task 'print_run_id[0]': Finished task run for task with final state: 'Success'
└── 19:44:42 | INFO    | Task 'print_run_id[1]': Starting task run...
└── 19:44:42 | INFO    | prefect.context['task_run_id']='dbc2e576-654a-46da-a4d5-ae4f6d15aef5', prefect.context['map_index']=1
└── 19:44:42 | INFO    | Task 'print_run_id[1]': Finished task run for task with final state: 'Success'
└── 19:44:42 | INFO    | Flow run SUCCESS: all reference tasks succeeded
Flow run succeeded!
```

## Environment

`prefect==1.0.0`

**Comment by madkinsz** [comment]
Unfortunately, I do not know of a good way to resolve this for local runs. I ran into this with mapped `create_flow_run` calls locally and introduced a fix at #5443. I wanted to fix this lower-level issue and always generate a unique id for each local task run, but it's not small enough to prioritize right now.

**Comment by bnaul** [comment]
Makes sense, thanks @madkinsz I did see that linked issue but didn't read carefully enough to see that it's basically the same (funny that the PR just happened to get merged). Up to you whether you wanna leave this open as an invitation or close as won't fix, I think probably we will stick with the same workaround and not look into it further either.
